### DISCUSSION TITLE
Battery delta cell voltage

### DISCUSSION THREAD

**Discussion by vytya** [davidrapan/ha-solarman#354] [Q&A] [answered]
Just wondering if there is possibility to implement at least delta cell voltage or better each cell voltage for a better understanding of what is happening with cells?

**Comment by davidrapan** [comment, marked as answer]
There is one possibility to get more info about your batteries when you have Deye inverter /w Deye battery but never per cell values to my knowledge.

**Comment by sofkaski** [comment]
@vytya : What brand your battery and BMS are? There might be HA integrations that can talk to BMS and have information on cell level.  E.g. I have a Seplos BMS it gives cell level voltages and alarms in addition to battery pack level data. For example:

![image](https://github.com/user-attachments/assets/1bc72e39-fcdf-49eb-b1f8-9c5c14418886)

**Reply by vytya** [reply]
I have custom-built on JK BMS. 
I know that there is possibility to make some solution with additional ESP32, but thought that solarman is already grabs that data from bms.

**Reply by sofkaski** [reply]
Solarman is limited to BMS data that comes via the BMS - inverter communication (CAN-bus or RS-485) and what the inverter publishes as registers to modbus.
